src/main/scala/Application.scala

```diff
@@ -0,0 +1,56 @@
+import scala.io.Source
+import org.json.{XML, JSONObject, JSONArray}
+import java.net.URL
+
+object Application extends App {
+
+  def getTestSuiteObject(filepath: URL): JSONObject = {
+    val bufferedSource = Source.fromURL(filepath)
+    val xml = bufferedSource.mkString
+    bufferedSource.close
+    XML.toJSONObject(xml).getJSONObject("testsuite")
+  }
+
+  def getTestCasesJSON(filepath: URL): JSONArray = {
+    getTestSuiteObject(filepath).getJSONArray("testcase")
+  }
+
+  // log, not xml
+  def findErrorsInLog(logFilePath: URL): String = {
+    val fileSource = Source.fromURL(logFilePath)
+    val rawContent = fileSource.mkString
+    fileSource.close
+    if (rawContent.contains("[error] (Test / compileIncremental) Compilation failed")) rawContent else ""
+  }
+
+  def toExercismJSON(filepath: URL, logFilePath: URL): JSONObject = {
+    val baseObject = new JSONObject().put("version", 2)
+    val errorMessage = findErrorsInLog(logFilePath)
+    if(!errorMessage.isEmpty) {
+      baseObject
+      .put("status", "error")
+      .put("message", errorMessage)
+    } else {
+      val testSuite = getTestSuiteObject(filepath)
+      val failuresNum = testSuite.getInt("failures")
+      val testCasesArray = testSuite.getJSONArray("testcase")
+      // val fileSource = Source.fromURL(logFilePath)
+      Console.err.println("asdasd")
+      val testCases: Array[JSONObject] = (0 until testCasesArray.length).toArray.map(idx => {
+        val o = testCasesArray.getJSONObject(idx)
+        val fail = o.optJSONObject("failure")
+        new JSONObject()
+        .put("name", o.getString("name"))
+        .put("status", if(fail != null) "fail" else "pass" )
+        .put("message", if(fail != null) fail.getString("message") else "")
+        .put("output", "TOIMPLEMENT")
+        .put("test_code", "TOIMPLEMENT")
+      })
+
+      baseObject
+      .put("status", if(failuresNum > 0) "fail" else "pass")
+      .put("message", "")
```

Comment:
As per [the docs](https://github.com/exercism/docs/blob/main/anatomy/track-tooling/test-runners/interface.md#message):

> When the status is not `error`, either set the value to `null` or omit the key entirely.

I'd suggest to just omit the key entirely:

```suggestion
```

Reply:
I'd rather keep it - in FP it's best practice to always return some value, and moreover it's always the best, IMHO, to keep all the parameters, even if null. This way you avoid errors such as "the key is inexistent".

Reply:
> in FP it's best practice to always return some value

I don't see how removing this field affects that, as you're still returning a value 🤷 

> it's always the best, IMHO, to keep all the parameters, even if null.

I'm fine with that, as long as it null and not an empty string.

Reply:
sure, I will make it a `null` then :)